examples/fabric-sync/admin/DeviceSynchronization.cpp

```diff
@@ -81,58 +81,65 @@ void DeviceSynchronizer::OnAttributeData(const ConcreteDataAttributePath & path,
+            mCurrentDeviceData.vendorId = static_cast<chip::VendorId>(attribUInt32Value);
         }
-    }
-    break;
-    case Clusters::BasicInformation::Attributes::ProductName::Id: {
-        char productNameBuffer[kBasicInformationAttributeBufSize];
-        if (SuccessOrLog(data->GetString(productNameBuffer, sizeof(productNameBuffer)), "ProductName"))
+        break;
+    case Clusters::BasicInformation::Attributes::VendorName::Id:
+        if (SuccessOrLog(data->GetString(attribStringBuffer, sizeof(attribStringBuffer)), "VendorName"))
         {
-            mCurrentDeviceData.productName = std::string(productNameBuffer);
+            mCurrentDeviceData.vendorName = std::string(attribStringBuffer);
         }
-    }
-    break;
-    case Clusters::BasicInformation::Attributes::NodeLabel::Id: {
-        char nodeLabelBuffer[kBasicInformationAttributeBufSize];
-        if (SuccessOrLog(data->GetString(nodeLabelBuffer, sizeof(nodeLabelBuffer)), "NodeLabel"))
+        break;
+    case Clusters::BasicInformation::Attributes::ProductID::Id:
+        if (SuccessOrLog(data->Get(attribUInt32Value), "ProductID"))
         {
-            mCurrentDeviceData.nodeLabel = std::string(nodeLabelBuffer);
+            mCurrentDeviceData.productId = attribUInt32Value;
         }
-    }
-    break;
-    case Clusters::BasicInformation::Attributes::HardwareVersionString::Id: {
-        char hardwareVersionStringBuffer[kBasicInformationAttributeBufSize];
-        if (SuccessOrLog(data->GetString(hardwareVersionStringBuffer, sizeof(hardwareVersionStringBuffer)),
-                         "HardwareVersionString"))
+        break;
+    case Clusters::BasicInformation::Attributes::ProductName::Id:
+        if (SuccessOrLog(data->GetString(attribStringBuffer, sizeof(attribStringBuffer)), "ProductName"))
         {
-            mCurrentDeviceData.hardwareVersionString = std::string(hardwareVersionStringBuffer);
+            mCurrentDeviceData.productName = std::string(attribStringBuffer);
         }
-    }
-    break;
-    case Clusters::BasicInformation::Attributes::SoftwareVersionString::Id: {
-        char softwareVersionStringBuffer[kBasicInformationAttributeBufSize];
-        if (SuccessOrLog(data->GetString(softwareVersionStringBuffer, sizeof(softwareVersionStringBuffer)),
-                         "SoftwareVersionString"))
+        break;
+    case Clusters::BasicInformation::Attributes::NodeLabel::Id:
+        if (SuccessOrLog(data->GetString(attribStringBuffer, sizeof(attribStringBuffer)), "NodeLabel"))
         {
-            mCurrentDeviceData.softwareVersionString = std::string(softwareVersionStringBuffer);
+            mCurrentDeviceData.nodeLabel = std::string(attribStringBuffer);
         }
-    }
-    break;
+        break;
+    case Clusters::BasicInformation::Attributes::HardwareVersionString::Id:
+        if (SuccessOrLog(data->GetString(attribStringBuffer, sizeof(attribStringBuffer)), "HardwareVersionString"))
```

Comment:
We should not use HardwareVersionString to read HardwareVersionString since they have different size